src/js/components/DataTable/Header.js

```diff
@@ -398,7 +400,7 @@ const Header = forwardRef(
                     <Box
                       direction="row"
                       align="center"
-                      gap="xsmall"
+                      gap={theme.dataTable.sorter?.gap}
```

Comment:
```suggestion
                      gap={theme.dataTable.sort?.gap}
```
not sure if this naming would be better ?